capz/run-capz-e2e.sh

```diff
@@ -482,7 +492,8 @@ set_azure_envs() {
     # Generate SSH key.
     capz::util::generate_ssh_key
 
-    AZURE_LOCATION=$(get_random_region)
+    # Set the Azure Location, preferred locationc can be set through the AZURE_LOCATION environment variable.
+    AZURE_LOCATION="${AZURE_LOCATION:-$(capz::util::get_random_region)}"
```

Comment:
allowing its override works but let's use `get_random_region`. We do keep in sync with capz but have had to change it occasionally